feature-libs/quote/components/quote-list/quote-list.component.ts

```diff
@@ -31,18 +40,81 @@ export class QuoteListComponent {
     this.quoteListService.setCurrentPage(page);
   }
 
-  getQuoteStateClass(state: string): string {
+  protected getBuyerQuoteStatus(state: string): string | undefined {
     switch (state) {
-      case 'BUYER_DRAFT':
+      case QuoteState.BUYER_DRAFT:
         return 'quote-draft';
-      case 'BUYER_SUBMITTED':
+      case QuoteState.BUYER_SUBMITTED:
         return 'quote-submitted';
-      case 'BUYER_REJECTED':
+      case QuoteState.BUYER_ACCEPTED:
+        return 'quote-accepted';
+      case QuoteState.BUYER_APPROVED:
+        return 'quote-approved';
+      case QuoteState.BUYER_REJECTED:
         return 'quote-rejected';
-      case 'CANCELLED':
+      case QuoteState.BUYER_OFFER:
+        return 'quote-offer';
+      case QuoteState.BUYER_ORDERED:
+        return 'quote-ordered';
+    }
+  }
+
+  protected getSellerQuoteStatus(state: string): string | undefined {
+    switch (state) {
+      case QuoteState.SELLER_DRAFT:
+        return 'quote-draft';
+      case QuoteState.SELLER_SUBMITTED:
+        return 'quote-submitted';
+      case QuoteState.SELLER_REQUEST:
+        return 'quote-request';
+    }
+  }
+
+  protected getSellerApproverQuoteStatus(state: string): string | undefined {
+    switch (state) {
+      case QuoteState.SELLERAPPROVER_APPROVED:
+        return 'quote-approved';
+      case QuoteState.SELLERAPPROVER_REJECTED:
+        return 'quote-rejected';
+      case QuoteState.SELLERAPPROVER_PENDING:
+        return 'quote-pending';
+    }
+  }
+
+  protected getGeneralQuoteStatus(state: string): string | undefined {
+    switch (state) {
+      case QuoteState.CANCELLED:
         return 'quote-cancelled';
-      default:
-        return '';
+      case QuoteState.EXPIRED:
+        return 'quote-expired';
+    }
+  }
+
+  protected isResponsible(
+    responsiblePersonPrefix: string,
+    state: string
+  ): boolean {
+    if (state && state.indexOf(responsiblePersonPrefix) >= 0) {
+      return true;
+    }
+    return false;
+  }
+
+  getQuoteStateClass(state: string | undefined): string {
```

Comment:
Let's make state mandatory in OccQuote (because I cannot imagine a situation where there would be no state defined), would you agree?
Then state can here be required as well

Reply:
Agree!